promdump/promdump.go

```diff
@@ -368,7 +376,118 @@ func exportMetric(ctx context.Context, promApi v1.API, metric string, beginTS ti
 			}
 		}
 	}
-	return writeFile(&values, filePrefix, fileNum)
+
+	return fileNum, writeFile(&values, filePrefix, fileNum)
+
+}
+
+// Holds custom metric file counts
+var customMetricCount uint
+
+func createArchive(buf io.Writer) error {
+	// Create new Writers for gzip , bzip2 and none(tar)
+
+	var tw *tar.Writer
+
+	switch *tarCompression {
+	case "gzip":
+		gw := gzip.NewWriter(buf)
+		defer gw.Close()
+		tw = tar.NewWriter(gw)
+
+	case "bzip2":
+		bw, err := bzip2.NewWriter(buf, nil)
+		if err != nil {
+			return err
+		}
+		defer bw.Close()
+		tw = tar.NewWriter(bw)
+
+	case "none":
+		tw = tar.NewWriter(buf)
+	default:
+		fmt.Errorf("unsupported compression type: %s , using default tar_compression_algorithm", *tarCompression)
+
+	}
+	defer tw.Close()
+
+	// Iterate over collectMetrics and add them to the tar archive
+	for _, v := range collectMetrics {
+		for i := 0; i < int(v.fileCount); i++ {
+			metricName, err := getMetricName(v)
+			if err != nil {
+				log.Fatalf("createArchive: %v", err)
+			}
+			filename := fmt.Sprintf("%s.%05d", metricName, i)
+			err = addToArchive(tw, filename)
+			if err != nil {
+				return err
+			}
+
+		}
+	}
+	//custom filename collect and add them to the tar archive
+	if *out != "" {
+		for i := 0; i < int(customMetricCount); i++ {
+			filename := fmt.Sprintf("%s.%05d", *out, i)
+			err := addToArchive(tw, filename)
+			if err != nil {
+				return err
+			}
+
+		}
+
+	}
+
+	return nil
+}
+
+func addToArchive(tw *tar.Writer, filename string) error {
+	// Open the file which will be written into the archive
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Get FileInfo about our file providing file size, mode, etc.
+	info, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	// Create a tar Header from the FileInfo data
+	header, err := tar.FileInfoHeader(info, info.Name())
+	if err != nil {
+		return err
+	}
+
+	// Write file header to the tar archive
+	if err := tw.WriteHeader(header); err != nil {
+		return err
+	}
+
+	// Copy file content to tar archive
+	_, err = io.Copy(tw, file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func generateDefaultTarFilename() string {
+	compressionType := *tarCompression
+
+	switch compressionType {
```

Comment:
Why not just `switch *tarCompression`? The `tarCompression` var might be more intuitive as `tarCompressionType` or something like that.

Reply:
Removed  compressionType